format/SparseTensor.fbs

```diff
@@ -113,9 +113,71 @@ table SparseMatrixIndexCSX {
   indicesBuffer: Buffer;
 }
 
+/// Compressed Sparse Fiber (CSF) sparse tensor index.
+table SparseTensorIndexCSF {
+  /// CSF is a generalization of compressed sparse row (CSR) index.
+  /// See [smith2017knl]: http://shaden.io/pub-files/smith2017knl.pdf
+  ///
+  /// CSF recursively compresses each mode of the tensor into a set
+  /// of prefix trees. Each path from a root to leaf forms one tensor
+  /// non-zero index. CSF is implemented with two buffers and three arrays.
+  ///
+  /// For example, let X be a 3x3x3x4 tensor, and it has the following
+  /// 8 non-zero values:
+  ///
+  ///   X[1, 1, 1, 2] := 1
+  ///   X[1, 1, 1, 3] := 2
+  ///   X[1, 2, 1, 1] := 3
+  ///   X[1, 2, 1, 3] := 4
+  ///   X[1, 2, 2, 1] := 5
+  ///   X[2, 2, 2, 1] := 6
+  ///   X[2, 2, 2, 2] := 7
+  ///   X[2, 2, 2, 3] := 8
+  ///
+  /// As a prefix tree this would be represented be:
+  ///
+  ///         1          2
+  ///        / \         |
+  ///       1   2        2
+  ///      /   / \       |
+  ///     1   1   2      2
+  ///    / \ / \   \    /|\
+  ///   2  3 1  3   1  1 2 3
+
+  /// The type of values in indptrBuffers
+  indptrType: Int;
+
+  /// indptrBuffers stores the sparsity structure.
+  /// Position in the indptrBuffers vector signifies the dimension.
+  /// For example, the indptrBuffers for the above X is:
+  ///
+  ///   indptrBuffer(X) = [[0, 2, 3], [0, 1, 3, 4], [0, 2, 4, 5, 8]].
+  ///
+  indptrBuffers: [Buffer];
```

Comment:
Could you please mark `indptrBuffers`, `indicesBuffers`, and `axisOrder` property required?
You can see how to do this in https://github.com/apache/arrow/pull/6234.

Reply:
Done. I've also made `indptrType` and `indicesType` required since I see `SparseMatrixIndexCSX` and `SparseTensorIndexCOO` have it as required.